lib/components/screens/search/search_screen.dart

```diff
@@ -1,63 +1,24 @@
-import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:harpy/components/components.dart';
-import 'package:harpy/core/core.dart';
-import 'package:harpy/harpy_widgets/harpy_widgets.dart';
-import 'package:harpy/misc/misc.dart';
-import 'package:provider/provider.dart';
 
 class SearchScreen extends StatelessWidget {
-  const SearchScreen();
+  const SearchScreen({
+    Key? key,
+  }) : super(key: key);
 
-  Widget _buildUserSearchCard() {
-    return HarpyListCard(
-      leading: const Icon(CupertinoIcons.search),
-      title: const Text('users'),
-      onTap: () => app<HarpyNavigator>().pushNamed(UserSearchScreen.route),
-    );
-  }
-
-  Widget _buildTweetSearchCard() {
-    return HarpyListCard(
-      leading: const Icon(CupertinoIcons.search),
-      title: const Text('tweets'),
-      onTap: () => app<HarpyNavigator>().pushTweetSearchScreen(),
-    );
-  }
+  static const String route = 'search';
 
   @override
   Widget build(BuildContext context) {
-    final mediaQuery = MediaQuery.of(context);
-    final config = context.watch<ConfigCubit>().state;
-
-    return ScrollToStart(
-      child: CustomScrollView(
-        slivers: [
-          const HomeTopSliverPadding(),
-          SliverPadding(
-            padding: config.edgeInsets.copyWith(bottom: 0),
-            sliver: SliverList(
-              delegate: SliverChildListDelegate([
-                _buildUserSearchCard(),
-                defaultVerticalSpacer,
-                _buildTweetSearchCard(),
-              ]),
-            ),
-          ),
-          SliverToBoxAdapter(
-            child: Divider(height: config.paddingValue * 2),
-          ),
-          const SliverToBoxAdapter(
-            child: TrendsCard(),
-          ),
-          const SliverToBoxAdapter(
-            child: defaultVerticalSpacer,
-          ),
-          const TrendsList(),
-          SliverToBoxAdapter(
-            child: SizedBox(height: mediaQuery.padding.bottom),
-          ),
-        ],
+    return HarpyScaffold(
+      body: BlocProvider<TrendsBloc>(
+        create: (_) => TrendsBloc()..add(const FindTrendsEvent()),
```

Comment:
We already build a trends bloc in `HomeProvider` for the home screen, so we shouldn't build another one here.
Instead you can add the `TrendsBloc` as a parameter and build a `BlocProvider.value` here instead

Reply:
Done